src/Uno.Toolkit.UI/Controls/NavigationBar/NativeNavigationBarPresenter.Android.cs

```diff
@@ -40,8 +49,19 @@ public void SetOwner(NavigationBar navigationBar)
 		private void OnLoaded(object sender, RoutedEventArgs e)
 		{
 			var navBar = TemplatedParent as NavigationBar;
-			Content = navBar?.GetRenderer(() => new NavigationBarRenderer(navBar)).Native;
+			if (navBar is { })
+			{
+				Content = navBar.GetRenderer(() => new NavigationBarRenderer(navBar)).Native;
+				navBar.MainCommand.Click += OnMainCommandClicked;
+				_mainCommandClickHandler.Disposable = Disposable.Create(() => navBar.MainCommand.Click -= OnMainCommandClicked);
```

Comment:
Make sure to dispose the current one, in case we're registering the same instance.
```suggestion
				_mainCommandClickHandler.Disposable = null;
				_mainCommandClickHandler.Disposable = Disposable.Create(() => navBar.MainCommand.Click -= OnMainCommandClicked);
```